### DISCUSSION TITLE
Support off-line signing

### DISCUSSION THREAD

**Discussion by chapellu** [hyperledger/fabric-admin-sdk#90] [Ideas]
It would be really nice to be able to generate the config_update via the sdk and get the digest like it is feasible in fabric-gateway for proposal. This will enable us to do off-line signing

**Comment by bestbeforetoday** [comment]
Do you have a use-case where the application driving your admin updates to the network using this admin SDK is actually an intermediary acting on behalf of a remote client who actually holds the signing keys? Or to put it another way, a use-case where the person holding the signing keys is not able to locally run a client application that uses the admin SDK? I (perhaps naively) imagined that was an unlikely scenario for network administration, and that this would take place relatively capable devices, relatively close to the actual network nodes.

If you just need to be able to plug in alternative signing mechanisms, like for example a Hardware Security Module, you can do that by providing your own Signer implementation that calls out to the appropriate signing mechanism.
